pkg/model/model.go

```diff
@@ -0,0 +1,90 @@
+package model
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
```

Comment:
In other places we use `logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"` - any particular reason to keep both?

Reply:
logf.Log.WithName("controller_session")

returns a go-logr object. At this level we're interested in the log object, not how it's created.